Azure/sonic-buildimage
# DellEMC: S6100, S6000 - Enable thermalctld, Platform API implementation and fixes

**- Why I did it**

To incorporate the below changes in DellEMC S6100, S6000 platforms.

- S6100, S6000:
    - Enable 'thermalctld'
    - Implement DeviceBase methods (presence, status, model, serial) for Fantray and Component
    - Implement ‘get_position_in_parent’, ‘is_replaceable’ methods for all device types
    - Implement ‘get_status’ method for Fantray
    - Implement ‘get_temperature’, ‘get_temperature_high_threshold’, ‘get_voltage_high_threshold’, ‘get_voltage_low_threshold’ methods for PSU
    - Implement ‘get_status_led’, ‘set_status_led’ methods for Chassis
    - SFP:
        - Make EEPROM read both Python2 and Python3 compatible
        - Fix ‘get_tx_disable_channel’ method’s return type
        - Implement ‘tx_disable’, ‘tx_disable_channel’ and ‘set_power_override’ methods
- S6000:
    - Move PSU thermal sensors from Chassis to respective PSU
    - Make available the data of both Fans present in each Fantray


**- How I did it**

- Remove 'skip_thermalctld:true' in pmon_daemon_control.json
- Implement the platform API methods in the respective device files
- Use `bytearray` for data read from transceiver EEPROM 
- Change return type of 'get_tx_disable_channel' to match specification in sonic_platform_common/sfp_base.py

**- How to verify it**

Verified that platform data is displayed by `show platform fan` and `show platform temperature` commands.
Wrote a python script to load Chassis class and then call the APIs accordingly.

UT Logs: 
[S6000_platform_API_UT_logs.txt](https://github.com/Azure/sonic-buildimage/files/5807451/S6000_platform_API_UT_logs.txt)
[S6100_platform_API_UT_logs.txt](https://github.com/Azure/sonic-buildimage/files/5807458/S6100_platform_API_UT_logs.txt)

**- Which release branch to backport (provide reason below if selected)**




**- Description for the changelog**


DellEMC: S6100, S6000 - Enable thermalctld, Platform API implementation and fixes

**- A picture of a cute animal (not mandatory but encouraged)**
